vars/yamlSubstituteVariables.groovy

```diff
@@ -0,0 +1,314 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.GenerateDocumentation
+import com.sap.piper.variablesubstitution.ExecutionContext
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+private void debug(String message) {
```

Comment:
I would prefer to use the normal `verbose` flag. In case the `verbose` flag is set we should emit to the log. Changing configuration is easier compared to changing the code in case of e.g. a support case from a customer.

Reply:
Could you spend a bit more on explaining how to use the `verbose` feature.
I was looking for a better way to do this in your code (something like a logging framework) but could not find one quickly.
So if you let me know how it should be done, I am happy to change it.

Reply:
**Update** - It's done. I pushed a new commit that resolves this.